Tests/OTelTestingTests/HookedFatalErrorTests.swift

```diff
@@ -0,0 +1,56 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift OTel open source project
+//
+// Copyright (c) 2024 Moritz Lang and the Swift OTel project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+@testable import OTel
+@testable import OTelTesting
+import XCTest
+
+final class HookedFatalErrorTests: XCTestCase {
```

Comment:
I appreciate you adding this extra test target to increase code coverage, but I don't think we necessarily need it anymore. What do you think, now that `OTelTesting` is code-coverage-ignored anyways? I'm happy to merge this as is if you think it's still valuable to have this test case and given you already implemented the tests.